homework9/hw9task1/iterators_1/merge_sorted_files.py

```diff
@@ -0,0 +1,26 @@
+"""
+Write a function that merges integer from sorted files and returns an iterator
+file1.txt:
+1
+3
+5
+file2.txt:
+2
+4
+6
+#>>> list(merge_sorted_files(["file1.txt", "file2.txt"]))
+[1, 2, 3, 4, 5, 6]
+"""
+from heapq import merge
+from pathlib import Path
+from typing import Iterator, List, Union
+
+
+def fetching_data(file_path: str):
+    with open(file_path, "r") as file:
+        yield from (int(line.rstrip()) for line in file)
+
+
+def merge_sorted_files(file_list: List[Union[Path, str]]) -> Iterator:
+    all_data = (fetching_data(file) for file in file_list)
+    return iter(merge(*all_data))
```

Comment:
Do you need to wrap merge to iter?

Reply:
You are right, we don`t need to wrap merge() to iter(). As I understand, theoretically merge() returns a generator which is also some kind of Iterator, and here should be `yield from merge(*all_data)` instead of `return iter(merge(*all_data))`

Reply:
You are right, we dont need to wrap merge() to iter(). As I understand, theoretically merge() returns a generator which is also some kind of Iterator, and here should be `yield from merge(*all_data)`  instead of  `return iter(merge(*all_data))`

Reply:
Exactly. 
Keep an eye on that pattern.   When you wrap one iterable to another, you have done it with `list` before. Always ask yourself if you wrap a collection without modifying it, what does it give you.  Now after the lecture about iterators you have all information you need about it.